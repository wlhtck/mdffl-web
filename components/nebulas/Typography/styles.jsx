import { black, white } from '../colors';

const marginBottom = { margin: '0 0 32px' };
const colors = { white, black };
const families = { raleway: 'Raleway', lato: 'Lato' };

/** @function fontStyles
  * Get font styles
  * @param {Object} props - The font properties.
  * @param {('left'|'right'|'center')} [props.align='left'] - css property text-align
  * @param {('black'|'white')} [props.color='black'] - css property color (text color)
  * @param {('lato'|'raleway')} [props.family='lato'] - css property color (text color)
  * @param {string} [props.size='24px'] - css property font-size
  * @param {number} [props.weight=600] - css property font-weight */

export default ({
  align = 'left', color = 'black', family = 'lato', size = '24px', weight = 600,
} = {}) => ({
  color: colors[color],
  fontFamily: `'${families[family]}', sans-serif`,
  fontWeight: weight,
  fontSize: size,
  textAlign: align,
  ...marginBottom,
});
