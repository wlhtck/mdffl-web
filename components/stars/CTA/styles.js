import { mergeAll } from 'lodash/fp';
import { primary, white } from '../../nebulas/colors';
import fontStyles from '../../nebulas/Typography/styles';

const backgroundColors = { primary, secondary: white };
const fontColors = { primary: 'white', secondary: 'primary' };

const buttonStyles = ({ type = 'primary' } = {}) => mergeAll([
  fontStyles({
    family: 'raleway',
    size: '16px',
    weight: 800,
    color: fontColors[type],
    lineHeight: 1,
  }), {
    backgroundColor: backgroundColors[type],
    display: 'inline-block',
    padding: '16px',
    border: 'none',
    borderRadius: '4px',
    cursor: 'pointer',
    textDecoration: 'none',
  },
]);

export default buttonStyles;
