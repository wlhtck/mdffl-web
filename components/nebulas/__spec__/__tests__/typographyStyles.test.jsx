import styles from '../../Typography/styles';

describe('nebulas/Typography/styles', () => {
  it('returns the default styles', () => {
    expect(styles()).toMatchSnapshot();
  });

  it('returns the specified styles', () => {
    expect(styles({
      align: 'center', color: 'white', family: 'raleway', size: '36px', weight: 800,
    })).toMatchSnapshot();
  });
});
