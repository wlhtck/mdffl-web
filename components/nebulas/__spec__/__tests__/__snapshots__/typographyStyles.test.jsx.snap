// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nebulas/Typography/styles returns the default styles 1`] = `
Object {
  "color": "#303133",
  "fontFamily": "'Lato', sans-serif",
  "fontSize": "24px",
  "fontWeight": "normal",
  "lineHeight": "normal",
  "margin": "0 0 32px",
  "textAlign": undefined,
}
`;

exports[`nebulas/Typography/styles returns the specified styles 1`] = `
Object {
  "color": "#FFFFFF",
  "fontFamily": "'Raleway', sans-serif",
  "fontSize": "36px",
  "fontWeight": 800,
  "lineHeight": "normal",
  "margin": "0 0 32px",
  "textAlign": "center",
}
`;
